crates/core/tedge_api/src/workflow/state.rs

```diff
@@ -281,6 +374,104 @@ fn json_as_string(value: &Value) -> String {
     }
 }
 
+/// A set of values to be injected/extracted into/from a [GenericCommandState]
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq)]
+#[serde(try_from = "Option<Value>")]
+pub enum StateExcerpt {
```

Comment:
Some unit tests for the conversions to/from this struct would be good.

Reply:
You seem to have missed this. Can be added in a follow-up PR as well, to avoid blocking this one.